source/plugins/go/src/oms.go

```diff
@@ -1880,6 +1896,128 @@ func containsKey(currentMap map[string]bool, key string) bool {
 	return c
 }
 
+func writeMsgPackEntries(connection net.Conn, isContainerLogV2Schema bool, fluentForwardTag string, msgPackEntries []MsgPackEntry) (totalBytes int, err error) {
+	var bts int
+	var er error
+	if IsAzMonMultiTenancyLogCollectionEnabled && isContainerLogV2Schema && !IsGenevaLogsIntegrationEnabled {
+		namespaceStreamIdsMap, streamIdNamedPipeMap := getContainerLogV2ExtensionMaps()
+		if len(namespaceStreamIdsMap) > 0 {
+			MultitenantNamespaceCount = len(namespaceStreamIdsMap)
+			streamTagCount := 0
+			for _, streamTags := range namespaceStreamIdsMap {
+				streamTagCount += len(streamTags)
+			}
+			ContainerLogV2ExtensionDCRCount = streamTagCount
+			msgPackEntriesByNamespace := getMsgPackEntriesByNamespace(msgPackEntries)
+			totalBytes := 0
+			for namespace, entries := range msgPackEntriesByNamespace {
+				if streamTags, exists := namespaceStreamIdsMap[namespace]; exists {
+					msg := fmt.Sprintf("Info::ama:: namespace : %s streamTags: %s \n", namespace, strings.Join(streamTags, ", "))
+					Log(msg)
+					for _, streamTag := range streamTags {
+						msgpBytes := convertMsgPackEntriesToMsgpBytes(streamTag, entries)
+						deadline := 10 * time.Second
+						if IsWindows {
+							// in windows, there will be dedicated namedpipe for each DCR and hence use namedpipe specific to DCR
+							namedPipe, ok := streamIdNamedPipeMap[streamTag]
+							Log("Info::ama:: namedpipe: %s and streamTag: %s: namespace: %s \n", namedPipe, streamTag, namespace)
+							if ok {
+								namedPipeConn, ok := NamedPipeConnectionCache[namedPipe]
+								if !ok || namedPipeConn == nil {
+									CreateWindowsNamedPipeClient(namedPipe, &namedPipeConn)
+									NamedPipeConnectionCache[namedPipe] = namedPipeConn
```

Comment:
should this cache be added after checking write success?

Reply:
good catch. fixed.